lib/src/pie_menu_core.dart

```diff
@@ -122,12 +127,20 @@ class _PieMenuCoreState extends State<PieMenuCore>
   /// If the [PieMenu] does not have a theme, [PieCanvas] theme is used.
   PieTheme get _theme => widget.theme ?? _notifier.canvasTheme;
 
+  @override
+  void initState() {
+    widget.controller?.addListener(_onControllerChanged);
+    super.initState();
+  }
+
   @override
   void dispose() {
     _overlayFadeController.dispose();
     _bounceController.dispose();
     _debounceTimer?.cancel();
     _bounceStopwatch.stop();
+    widget.controller?.dispose();
```

Comment:
I think we shouldn't dispose the controller here since it's initialized outside by the user. We just need to remove the listener.

Reply:
Should be addressed now